xugrid/regrid/structured.py

```diff
@@ -21,56 +28,619 @@ class StructuredGrid1d:
+            flipped = False
+            side = "left"
+        else:
+            raise ValueError(f"{name} is not monotonic for array {obj.name}")
+
+        if bounds_name in obj.coords:
+            bounds = obj[bounds_name].values
+        else:
+            if size_name in obj.coords:
+                # works for scalar size and array size
+                size = np.abs(obj[size_name].values)
+            else:
+                # no bounds defined, no dx defined
+                # make an estimate of cell size
+                size = np.diff(midpoints)
+                # Check if equidistant
+                atolx = 1.0e-4 * size[0]
+                if not np.allclose(size, size[0], atolx):
+                    raise ValueError(
+                        f"DataArray has to be equidistant along {name}, or "
+                        f'explicit bounds must be given as "{name}bounds", or '
+                        f'cellsizes must be as "d{name}"'
+                    )
+
+            start = midpoints - 0.5 * size
+            end = midpoints + 0.5 * size
+            bounds = np.column_stack((start, end))
+
+        self.name = name
+        self.midpoints = midpoints
         self.bounds = bounds
+        self.flipped = flipped
+        self.side = side
+        self.grid = obj
+
+    @property
+    def ndim(self):
+        return 1
+
+    @property
+    def dims(self):
+        return (self.name,)
 
     @property
     def size(self):
         return len(self.bounds)
 
-    def overlap(self, other, relative: bool):
+    @property
+    def length(self):
+        return abs(np.diff(self.bounds, axis=1))
+
+    def flip_if_needed(self, index):
+        if self.flipped:
+            return self.size - index - 1
+        else:
+            return index
+
+    def valid_nodes_index(self, other):
+        """
+        Retruns all nodes that are within bounding box of overlaying grid.
+        In cases that midpoints (and bounding boxes) are flipped, computed indexes
+        are fliped as well.
+
+        Args:
+            self (StructuredGrid1d): source grid
+            other (StructuredGrid1d): target grid
+
+        Returns:
+            valid_self_index (np.array): valid source indexes
+            valid_other_index (np.array): valid target indexes
+        """
+        start = np.searchsorted(self.bounds[:, 0], other.midpoints, side=self.side)
+        end = np.searchsorted(self.bounds[:, 1], other.midpoints, side=self.side)
+        valid = (
+            (start == (end + 1))
+            & (other.midpoints > self.bounds[0, 0])
+            & (other.midpoints < self.bounds[-1, 1])
+        )
+        valid_other_index = np.arange(other.size)[valid]
+        valid_self_index = end[valid]
+        valid_self_index = self.flip_if_needed(valid_self_index)
+        valid_other_index = other.flip_if_needed(valid_other_index)
+        return valid_self_index, valid_other_index
+
+    def valid_lineair_nodes_index(self, other):
+        """
```

Comment:
I get that you are returning the indexes of source and target that are valid for linear interpolation, but the function name is a bit confusing. (what is a valid linear node? )

Now the function name refers to what you want to do with the result (linear interpolation) but I think it would be nicer if it referred to geometrical concepts. Same for ohter functions in this class. If I see " valid_nodes_index" I don't know what it does but if I see "node_index_in_overlap_area" I get an idea what it does.